tests/test_sockets.py

```diff
@@ -1849,3 +1854,60 @@ async def test_connect_tcp_getaddrinfo_context() -> None:
             pass
 
     assert exc_info.value.__context__ is None
+
+
+@pytest.mark.parametrize("socket_type", ["socket", "fd"])
+@pytest.mark.parametrize("event", ["readable", "writable"])
+async def test_wait_socket(
+    anyio_backend_name: str, event: str, socket_type: str
+) -> None:
+    if anyio_backend_name == "asyncio" and sys.platform == "win32":
+        import asyncio
+
+        policy = asyncio.get_event_loop_policy()
+        if policy.__class__.__name__ == "WindowsProactorEventLoopPolicy":
+            pytest.skip("Does not work on asyncio/Windows/ProactorEventLoop")
+
+    wait = wait_readable if event == "readable" else wait_writable
+
+    def client(port: int) -> None:
+        with socket.socket(socket.AF_INET, socket.SOCK_STREAM) as sock:
+            sock.connect(("127.0.0.1", port))
+            sock.sendall(b"Hello, world")
+
+    with socket.socket(socket.AF_INET, socket.SOCK_STREAM) as sock:
+        sock.bind(("127.0.0.1", 0))
+        port = sock.getsockname()[1]
+        sock.listen()
+        thread = Thread(target=client, args=(port,), daemon=True)
+        thread.start()
+        conn, addr = sock.accept()
+        with conn:
+            sock_or_fd: HasFileno | int = conn.fileno() if socket_type == "fd" else conn
+            with fail_after(10):
+                await wait(sock_or_fd)
+                assert conn.recv(1024) == b"Hello, world"
+        thread.join()
+
+
+async def test_deprecated_wait_socket(anyio_backend_name: str) -> None:
+    if anyio_backend_name == "asyncio" and sys.platform == "win32":
```

Comment:
I prefer the use of `platform.system` in this code base.
```suggestion
    if anyio_backend_name == "asyncio" and platform.system() == "Windows"
```